### PR TITLE
Confirm that a newly asked subquestion is in Redux before snapshot

### PR DESCRIPTION
**Pull request by zjmiller**

### CODE REVIEW

File: client/src/components/NewBlockForm.tsx [hunk 1]
```diff
@@ -65,8 +65,11 @@ export class NewBlockFormPresentational extends React.Component<any, any> {
     };
   }
 
-  public componentWillReceiveProps() {
-    if (this.state.pending) {
+  public componentWillReceiveProps(newProps) {
+    if (
+      this.state.pending &&
+      newProps.numBlocksInRedux > this.props.numBlocksInRedux // This checks whether the newly asked sub-question has made it into Redux
```
Comment (andrewschreiber) [inline client/src/components/NewBlockForm.tsx]: This seems fine as long as Redux will trigger a new props refresh, which it should.

Do you think the `setState` on 81 should be moved out of this block?

File: client/src/components/NewBlockForm.tsx [hunk 2]
```diff
@@ -388,11 +391,15 @@ export class NewBlockFormPresentational extends React.Component<any, any> {
...
+const mapStateToProps = state => ({
+  numBlocksInRedux: state.blocks.blocks.length,
+});
+
 const mapDispatchToProps = dispatch => ({
   resetBlock: ({ id }) => dispatch(resetBlock({ id })),
 });
 
 export const NewBlockForm = connect(
-  undefined,
```
Comment (andrewschreiber) [inline client/src/components/NewBlockForm.tsx]: Curious how we think about `undefined` versus `null`. Notice we use `null` ~284 times and `undefined` ~33 times.

Comment (zjmiller) [reply client/src/components/NewBlockForm.tsx]: Yeah good question... Different people having different preferences. Might be something to keep an eye on in future PRs. Let me know if you have a preference for consistency.

Comment (andrewschreiber) [reply client/src/components/NewBlockForm.tsx]: Looked into this a bit. I think we should use `null`.
image
It seems in idiomatic javascript, when assigning an empty property value, null is preferred over undefined.
https://stackoverflow.com/questions/5076944/what-is-the-difference-between-null-and-undefined-in-javascript

**Comment by andrewschreiber** [comment]
Do you want a re-review here?

**Pull request merged**